### PR TITLE
conprof: fix genjidb(badger) gc doesn't work problem

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

close: #120

Related issue: https://github.com/genjidb/genji/issues/454

https://discuss.dgraph.io/t/badgerdb-consume-too-much-disk-space/17070

Related repo: https://github.com/crazycs520/genji/tree/v0.13.0.bugfix

### What is changed and how it works?

The bug was caused by genjidb, but genjidb doesn't support badger now, so I have to fork the genjidb then fix in my repo, then replace it in `go.mod`

### CODE REVIEW

File: component/conprof/store/store_test.go [hunk 1]
```diff
@@ -239,6 +243,78 @@ func TestGenjiDBAddColumn(t *testing.T) {
...
+	require.NoError(t, err)
+
+	// Insert 1000 kv to make sure the deleted kv will been compacted.
+	for i := 1000; i < 2000; i++ {
+		data := mockProfile()
+		sql = "INSERT INTO t (ts, data) VALUES (?, ?)"
+		err = db.Exec(sql, int(ts)+i, data)
+		require.NoError(t, err)
+	}
+
```
Comment (zhongzc) [inline component/conprof/store/store_test.go:292]: how about checking key count here?

File: database/timeseries/syscall_not_linux_unix.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build !linux
 // +build !linux
```
Comment (zhongzc) [inline database/timeseries/syscall_not_linux_unix.go:2]:
```suggestion
//go:build !linux
```

File: component/conprof/store/store_test.go (hunk 1 above)
Comment (crazycs520) [reply component/conprof/store/store_test.go:292]: I use `Flatten` API to make sure the kv will be compacted then the deleted kv will be delete when compaction.

If I check the key count here, I can't make sure the compaction whether happend.

File: database/timeseries/syscall_not_linux_unix.go (hunk 2 above)
Comment (crazycs520) [reply database/timeseries/syscall_not_linux_unix.go:2]: No, can't remove it...

File: go.sum [hunk 3]
```diff
@@ -215,6 +215,8 @@ github.com/cosiner/argv v0.1.0/go.mod h1:EusR6TucWKX+zFgtdUsKT2Cvg45K5rtpCcWz4hK
 github.com/cpuguy83/go-md2man v1.0.10/go.mod h1:SmD6nW6nTyfqj6ABTjUi3V3JVMnlJmwcJI5acqYI6dE=
 github.com/cpuguy83/go-md2man/v2 v2.0.0-20190314233015-f79a8a8ca69d/go.mod h1:maD7wRr/U5Z6m/iR4s+kqSMx2CaBsrgA7czyZG/E6dU=
 github.com/cpuguy83/go-md2man/v2 v2.0.1/go.mod h1:tgQtvFlXSQOSOSIRvRPT7W67SCa46tRHOmNcaadrF8o=
+github.com/crazycs520/genji/engine/badgerengine v0.12.1-0.20220328082424-727a2d089bde h1:URhoDN38NcI0UfTnCQZQ6h1254HPAlf/rDdwCDz03dI=
```
Comment (breeswish) [inline go.sum:218]: looks like your fix is based on 0.12.1 instead of 0.13.0. Is it fine?

Comment (crazycs520) [reply go.sum:218]: looks weird, my fix is base on v0.13.0, the hash is right: `727a2d089bde`

image

**Comment by breeswish** [comment]
@crazycs520 There is still no CI for ngm :/

**Review by breeswish — APPROVED** [review]

**Review by zhongzc — APPROVED** [review]

**Pull request merged**